kedro/extras/datasets/api/api_dataset.py

```diff
@@ -21,83 +21,66 @@ class APIDataSet(AbstractDataSet):
     """
 
-    # pylint: disable=too-many-arguments
     def __init__(
         self,
         url: str,
         method: str = "GET",
-        data: Any = None,
-        params: Dict[str, Any] = None,
-        headers: Dict[str, Any] = None,
-        auth: Union[Iterable[str], AuthBase] = None,
-        json: Union[List, Dict[str, Any]] = None,
-        timeout: int = 60,
-        credentials: Union[Iterable[str], AuthBase] = None,
+        load_args: Dict[str, Any] = None,
+        credentials: Iterable[str] = None,
     ) -> None:
         """Creates a new instance of ``APIDataSet`` to fetch data from an API endpoint.
 
         Args:
             url: The API URL endpoint.
             method: The Method of the request, GET, POST, PUT, DELETE, HEAD, etc...
-            data: The request payload, used for POST, PUT, etc requests
-                https://requests.readthedocs.io/en/master/user/quickstart/#more-complicated-post-requests
-            params: The url parameters of the API.
-                https://requests.readthedocs.io/en/master/user/quickstart/#passing-parameters-in-urls
-            headers: The HTTP headers.
-                https://requests.readthedocs.io/en/master/user/quickstart/#custom-headers
-            auth: Anything ``requests`` accepts. Normally it's either ``('login', 'password')``,
-                or ``AuthBase``, ``HTTPBasicAuth`` instance for more complex cases. Any
-                iterable will be cast to a tuple.
-            json: The request payload, used for POST, PUT, etc requests, passed in
-                to the json kwarg in the requests object.
-                https://requests.readthedocs.io/en/master/user/quickstart/#more-complicated-post-requests
-            timeout: The wait time in seconds for a response, defaults to 1 minute.
-                https://requests.readthedocs.io/en/master/user/quickstart/#timeouts
-            credentials: same as ``auth``. Allows specifying ``auth`` secrets in
-                credentials.yml.
-
+            load_args: Additional parameters to be fed to requests.request.
+                https://docs.python-requests.org/en/latest/api/
+            credentials: Allows specifying secrets in credentials.yml.
         Raises:
-            ValueError: if both ``credentials`` and ``auth`` are specified.
+            ValueError: if both ``auth`` in ``load_args`` and ``credentials`` are specified.
         """
         super().__init__()
 
-        if credentials is not None and auth is not None:
-            raise ValueError("Cannot specify both auth and credentials.")
+        self._load_args = load_args or {}
 
-        auth = credentials or auth
+        if credentials is not None and "auth" in self._load_args:
+            raise ValueError("Cannot specify both auth and credentials.")
 
-        if isinstance(auth, Iterable):
-            auth = tuple(auth)
+        self._auth = None
+        if credentials is not None:
+            self._auth = tuple(credentials)
```

Comment:
As I understand it, now there are two options for supplying credentials/auth:
1. if it's `auth` in `load_args` then it's assumed to be `AuthBase`. No casting to `tuple` in the case that it's an `Iterable`.
2. if it's `credentials` then it's assumed to be `Iterable` and we cast to `tuple`

This is a little different from now where `auth` and `credentials` can both be either `AuthBase` or `Iterable`. Hence it's a bit more of a breaking change than just moving things to `load_args`. I think this is fine (and an improvement on how we currently do it), but let's also mention it in docs/release notes. I'll make separate comments with suggestions for this.

Reply:
I have reverted the implementation and added more tests: `AuthBase` instances can be provided now.